server/lib/enrich/video.js

```diff
@@ -15,7 +15,10 @@ module.exports = exports = function video(content, contract, graphicSyndicationF
 	content.extension = DOWNLOAD_ARCHIVE_EXTENSION;
 
 	content.download = Array.from(content.attachments).reverse().find(item => item.mediaType === DOWNLOAD_MEDIA_TYPES.video);
-	content.download.extension = (content.download) ? mime.extension(content.download.mediaType) : '';
+
+	if (content.download) {
+		content.download.extension = (content.download) ? mime.extension(content.download.mediaType) : '';
+	}
```

Comment:
[Observation] I need to admit, I'm not sure what happens with `content.download.extension`. Because currently you only set the extension if the `content.download` is true, and we'd end up with `content.download.extension` of `undefined` which may or may not be a problem - I don't know off hand. 

Looking at this code, I'm a little bit confused as to how it works. So `content.download` is an array, right? But in the next line (in the old version) we're treating `content.download` as an object - so which one do we want it to be - an array, or an object?

Reply:
`content-attachments` is an array that can be empty (which currently causes an error) or, if populated, will look something like this:

```js
[
  {
    "url": "https://next-media-api.ft.com/renditions/15053038942430/640x360.mp4",
    "width": 640,
    "height": 360,
    "duration": 178943,
    "codec": "h264",
    "mediaType": "video/mp4"
  },
  {
    "url": "https://next-media-api.ft.com/renditions/15053038942430/1280x720.mp4",
    "width": 1280,
    "height": 720,
    "duration": 178943,
    "codec": "h264",
    "mediaType": "video/mp4"
  },
  {
    "url": "https://next-media-api.ft.com/renditions/15053038942430/1920x1080.mp4",
    "width": 1920,
    "height": 1080,
    "duration": 178943,
    "codec": "h264",
    "mediaType": "video/mp4"
  },
  {
    "mediaType": "text/vtt",
    "url": "https://next-media-api.ft.com/captions/15053038942430.vtt"
  }
]
```

`content.download`, providing a suitable object in the array can be found, will be an object (both before and after these changes).

Ref. [MDN Web Docs: Array.prototype.find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)
  > The `find()` method returns the first element in the provided array that satisfies the provided testing function. If no values satisfy the testing function, [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) is returned.

Reply:
Agh, yes, I was looking at the `Array.from`. Chaining has its place, but can also be confusing! 😅